src/main/java/com/synopsys/integration/detect/lifecycle/run/step/IntelligentModeStepRunner.java

```diff
@@ -70,6 +74,10 @@ public void runOffline(NameVersion projectNameVersion, DockerTargetData dockerTa
             IacScanStepRunner iacScanStepRunner = new IacScanStepRunner(operationRunner);
             iacScanStepRunner.runIacScanOffline();
         });
+        if (configurationFactory.componentLocationAnalysisEnabled()) {
```

Comment:
Since operation runner is already being passed to both `IntelligentModeStepRunner` and `RapidModeStepRunner` classes, and it already contains an instantiated `detectConfigurationFactory`, it'd be better to add a method in `OperationRunner` that takes this decision. This will eliminate the need of passing `configurationFactory` to both step runners & modifying constructors of both step runners for a boolean.
Example:
`operationRunner.shouldInvokeComponentLocator()` can be the condition here.
In OperationRunner, you can simply wrap a method around `detectConfigurationFactory.componentLocationAnalysisEnabled()`

Reply:
Perfect, done! Thanks